zerver/lib/import_realm.py

```diff
@@ -916,6 +916,36 @@ def create_users(realm: Realm, name_list: Iterable[Tuple[str, str]],
     bulk_create_users(realm, user_set, bot_type)
 
 def update_message_foreign_keys(import_dir: Path) -> None:
+    old_id_list = get_incoming_message_ids(import_dir=import_dir)
+
+    count = len(old_id_list)
+
+    new_id_list = allocate_ids(model_class=Message, count=count)
+
+    for old_id, new_id in zip(old_id_list, new_id_list):
+        update_id_map(
+            table='message',
+            old_id=old_id,
+            new_id=new_id,
+        )
+
+    # We don't touch user_message keys here; that happens later when
+    # we're actually read the files a second time to get actual data.
+
+def get_incoming_message_ids(import_dir: Path) -> List[int]:
```

Comment:
For this commit, can we just add a flag to the `realm.json`  "sort_messages_by_timestamp", and then have this re-sorting code not run at all for the Zulip->Zulip import process?  I don't like the risk of tiebreaks being done wrong for that case, where wer know for a fact the precise ordering we want.